src/agent/Agent.ts

```diff
@@ -136,24 +139,36 @@ export class Agent {
       await this.inboundTransporter.start(this)
     }
 
+    await this.mediationRecipient.init(this.connections)
+    const defaultMediator = await this.mediationRecipient.getDefaultMediatorConnection()
+    if (defaultMediator) {
+      // TODO: update with batch pickup protocol.
+      this.connections.pingMediator(defaultMediator)
+    }
     this._isInitialized = true
   }
 
   public get publicDid() {
     return this.wallet.publicDid
   }
-
-  public getMediatorUrl() {
-    return this.agentConfig.mediatorUrl
+  public async preparePackMessage(outboundMessage: OutboundMessage) {
```

Comment:
In the Newly added `WsInboundTransporter` there is a need to send a packed message over the WebSocket. To send a message from an agent, the agent wraps outbountTransporter `sendMessage`, and does not provide a way to send a message on an `InboundTransporter` or support multiple transport types given a service doc. To avoid hardcoded checks in the outbound transport I exposed `preparePackMessage` which allows an inbound transport to not only receive messages but to pack them for travel as well. This will not be needed when we add support for multiple transports given a service doc. 
Should I be injecting into the transporters to remove the need for this method?

Reply:
I see. I think the approach is fine, but would be nice to use injection instead

Reply:
we are now injecting the message sender inside of `WsInboundTransporter` and preparePackMessage has been removed from the agent.